app/src/main/java/com/cmput301w23t40/capturetheqr/FirstTimeLogInActivity.java

```diff
@@ -0,0 +1,60 @@
+package com.cmput301w23t40.capturetheqr;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.os.Bundle;
+import android.provider.Settings;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+public class FirstTimeLogInActivity extends AppCompatActivity {
+    private EditText editTextUsername;
+    private EditText editTextContactInfo;
+    private Button buttonSubmit;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_first_time_log_in);
+        editTextUsername = findViewById(R.id.edtxt_username);
+        editTextContactInfo = findViewById(R.id.edtxt_contactInfo);
+        buttonSubmit = findViewById(R.id.btn_submit);
+        buttonSubmit.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                createNewUser();
+            }
+        });
+    }
+
+    public void createNewUser(){
+        while(editTextUsername.toString().length() < 6 ||
+                whetherUsernameExists(editTextUsername.toString())){
+            // FIXME need to let the user know the reason of failure
+            // FIXME prompt to input another username
+        }
+        saveUser(editTextUsername.toString(), editTextContactInfo.toString(), getUserIdentifier(this));
+    }
+
+    public boolean whetherUsernameExists(String username){
+        // FIXME DB queries
+        return false;
+    }
+
+    public void saveUser(String username, String contactInfo, String deviceID){
+        // FIXME DB queries
+    }
+
+    // FIXME citation
```

Comment:
@real-rookie  Please cite similar to this:

    /* Adapted/Copied code from the following resource for the swipe-to-delete functionality: 

            author: https://auth.geeksforgeeks.org/user/chaitanyamunje

            url: https://www.geeksforgeeks.org/swipe-to-delete-and-undo-in-android-recyclerview/

            last updated: Oct 5, 2021

    *

Reply:
Thanks, will do, also according to the requirements of assignment 1, we also need to state the licence.

Reply:
oh true, I can create an issue for stating licences since we didn't do that for any of our citations yet

Reply:
issue is created here: #46